### PR TITLE
11 get api review queries

### PR DESCRIPTION
**Pull request by JamesATeasdale**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -308,3 +308,111 @@ describe("GET /api/users should return a list of all users", () => {
...
+				.expect(404)
+				.then(({ body }) => {
+					expect(body).toEqual({ msg: "Not found" });
+				}),
+			request(app)
+				.get("/api/reviews?category=7")
+				.expect(404)
+				.then(({ body }) => {
+					expect(body).toEqual({ msg: "Not found" });
+				}),
```
Comment (johnhutcheon) [inline __tests__/app.test.js:421]: Great testing so far James! Could I just request one more please...

- Status 200, valid category query, but has no reviews responds with an empty array of articles, e.g. ?category=children's games.

File: controllers/app.controller.js [hunk 2]
```diff
@@ -19,11 +19,32 @@ exports.getReview = (req, res, next) =>
...
+	const sortByWhiteList = [
+		"title",
+		"designer",
+		"owner",
+		"review_img_url",
+		"review_id",
+		"category",
+		"created_at",
+		"votes",
+	];
```
Comment (johnhutcheon) [inline controllers/app.controller.js]: We'd usually expect this to be in the model, then we can promise.reject if the sort by doesn't contain any of these.

**Review by johnhutcheon — CHANGES_REQUESTED** [review]
Hi James, nice work! There's just a couple more tests we need and a bit of code that should be moved from the controller into the model. If you're struggling at all, drop us an nchelp and someone can assist 👍

**Review by Pipe-Smoking-Rabbit — APPROVED** [review]
Looks ready to merge ✅  Very thorough testing I must say!

**Pull request merged**